nautobot/tenancy/views.py

```diff
@@ -78,29 +80,40 @@ class TenantListView(generic.ObjectListView):
 
 class TenantView(generic.ObjectView):
     queryset = Tenant.objects.select_related("tenant_group")
-
-    def get_extra_context(self, request, instance):
-        stats = {
-            "circuit_count": Circuit.objects.restrict(request.user, "view").filter(tenant=instance).count(),
-            "cluster_count": Cluster.objects.restrict(request.user, "view").filter(tenant=instance).count(),
-            "device_count": Device.objects.restrict(request.user, "view").filter(tenant=instance).count(),
-            "ipaddress_count": IPAddress.objects.restrict(request.user, "view").filter(tenant=instance).count(),
-            # TODO: Should we include child locations of the filtered locations in the location_count below?
-            "location_count": Location.objects.restrict(request.user, "view").filter(tenant=instance).count(),
-            "prefix_count": Prefix.objects.restrict(request.user, "view").filter(tenant=instance).count(),
-            "rack_count": Rack.objects.restrict(request.user, "view").filter(tenant=instance).count(),
-            "rackreservation_count": RackReservation.objects.restrict(request.user, "view")
-            .filter(tenant=instance)
-            .count(),
-            "dynamicgroup_count": DynamicGroup.objects.restrict(request.user, "view").filter(tenant=instance).count(),
-            "virtualmachine_count": VirtualMachine.objects.restrict(request.user, "view")
-            .filter(tenant=instance)
-            .count(),
-            "vlan_count": VLAN.objects.restrict(request.user, "view").filter(tenant=instance).count(),
-            "vrf_count": VRF.objects.restrict(request.user, "view").filter(tenant=instance).count(),
-        }
-
-        return {"stats": stats, **super().get_extra_context(request, instance)}
+    object_detail_content = ObjectDetailContent(
+        panels=(
+            ObjectFieldsPanel(
+                section=SectionChoices.LEFT_HALF,
+                weight=100,
+                fields=[
+                    "tenant_group",
+                    "description",
+                ],
+            ),
+            StatsPanel(
+                label="Stats",
+                filter_name="tenant",
+                filter_pks=[],
+                related_field_names={
+                    Circuit: "tenant__in",
```

Comment:
Might be nice if we could do either or both of:

- automatically append "__in" to the filter string in all cases?
- use either the model or the `filter_name` to derive a default filter string?

Just trying to think if we could do something even simpler on the declaration side, something like `related_models = [Circuit, Cluster, Device, IPAddress, Location, ....]`

Reply:
I like the pattern you suggested too. But I think when it comes to more complex queries in LocationDetailView, this current pattern might be more desirable.
For example:
```
"circuit_count": Circuit.objects.restrict(request.user, "view")
.filter(circuit_terminations__location__in=related_locations)
.count(),
"vm_count": VirtualMachine.objects.restrict(request.user, "view")
.filter(cluster__location__in=related_locations)
.count(),
```

Reply:
Yeah, I think we'd need to support both - maybe something like `related_models = [Device, Prefix, (Circuit, "circuit_terminations__location"), (VirtualMachine, "cluster__location"), ...]` might work and still be reasonably comprehensible to other developers?